### ISSUE TITLE
Transaction remains "pending" unless account is reset before it's triggered

### ISSUE THREAD

**Issue by wbt** [MetaMask/metamask-extension#7616]
**Describe the bug**
After changing private networks without resetting an account, a transaction from that account remains in the "Pending" state indefinitely, with no clue to the user about why. 

**To Reproduce**
I don't yet have this boiled down to a minimal reproducible example, but here's one condition that led me to observe the bug. 

SETUP: 

1. Two environments, with two independent blockchains exposed at two different RPC URLs (one each).
2. Account 1 has a great deal of ETH on both chains, originating from the genesis block.
3. The user uses MetaMask to create Account 2, which has nothing.
4. The user wishes to transfer some ETH from Account 1 to Account 2.  
5. On Network A, the user goes into MetaMask, where Account 1 is selected as current, and initiates a Send to Account 2, enters the amount of the transfer, and confirms.  
6. The transaction works as expected, and quickly becomes Confirmed in the history.
7. The user switches to Account 2 and triggers some other small transactions from it, all of which are confirmed correctly.
8. The user changes the network selector to the RPC URL for Network B.
9. The user changes the account selector back to Account 1.  
10. The user follows the same steps to initiate an ETH transfer to Account 2, on Network B.
11. The transaction remains Pending indefinitely, with no indication of why.
![Screenshot of pending transaction](https://user-images.githubusercontent.com/563406/69992129-0923e580-1517-11ea-9749-43aba3a16505.png)
12. After several minutes, the user clicks to see transaction details, and tries to Speed Up the transaction, doubling the gas price.
13. The transaction remains Pending indefinitely, with no indication of why.
14. After several minutes, the user clicks to Cancel the transaction, and agrees to the gas fee.
15. The transaction remains Pending indefinitely, with no indication of why.
16. The user switches the network selector back to Network A.
17. The transaction state changes to Cancelled after about a minute.
18. Still on Network A, the user initiates another ETH transaction from Account 1 to Account 2. 
19. That transaction is quickly confirmed.
20. The user switches the network selector back to the RPC address for Network B.   The history of transactions is displayed, but Account 2 has 0 ETH, as expected.
21. On Network B, the user initiates an ETH transaction from Account 1 to Account 2. 
22. The transaction remains Pending indefinitely, with no indication of why.
23. After several minutes, the user changes the network selector to the RPC address for Network A.
24. After about a minute, the transaction becomes Confirmed.  Account 2's balance is updated again on Network A.
25. The user switches back to the RPC address for Network B.
26. The user goes into Settings->Advanced and does Reset Account.
27. Still on Network B, the user triggers a transaction to send ETH from Account 1 to Account 2, still using MetaMask's interface to do so.
28. The transaction is quickly confirmed. 

**Expected behavior**
Initially: Steps 11-27 inclusive are skipped.
Mainly: Better communication to the user about why the transaction remained in a perpetually pending state, that might help them figure out the needed Reset Account fix.
Possibly: The needed nonce-adjustment action being taken automatically. 

**Browser details (please complete the following information):**
 - OS: Win10 Pro
 - Browser: Chrome 78.0.3904.108 (Official Build) (64-bit)
 - MetaMask Version: 7.6.1

**Additional context**
This *might* be a more carefully worded duplicate of #7607, which would satisfy its "N02-needsReproduction" label.  However, I was not seeing the specific error text reported there (and did not specifically notice anything in the browser console), so I did not file this as a comment on that.

**Comment by bschorchit** [comment]
Closing as this doesn't seem to be an issue anymore.

**Comment by wbt** [comment]
Is that because somebody tested it and verified that it's working, or because of a known code change to better keep track of which networks transactions are pending on, or because of a UI change to not display pending transactions for non-current networks, or because there haven't been a large number of reports of this network-switching slow-transaction-confirmation edge case?
